### PR TITLE
Automate gradle wrapper upgrade

### LINKED ISSUE #2449: Upgrade Gradle wrapper and automate it
The current Gradle version is quite old (7.5). This ticket should update the Gradle wrapper as well as adding an automation to update it as soon as a new release is available.
We can make use of [this plugin](https://plugins.gradle.org/plugin/org.gradle.wrapper-upgrade) for it.

### PR DESCRIPTION
**Pull request by StefMa**
## Pull Request (PR) Checklist
Please check if your pull request fulfills the following requirements:
- [x] The PR is submitted to the `main` branch.
- [x] I've read the **[Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md)**.
- [ ] The code builds and is tested on a real Android device.
- [x] I confirm that I've run the code locally and everything works as expected.

_Put an `x` in the boxes that apply._
- [x] Demo: Checking checkbox using `[x]`

> Upload a screen recording + screenshots to prove that your changes work.

## Pull Request Type
Please check the type of the changes that your PR introduce:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, new lines, etc.)
- [ ] Refactoring (no functional changes, renaming)
- [ ] Small improvement (fix typo, UI fine-tune or something small)
- [x] Gradle Build related changes
- [ ] Dependencies update (updating libraries)
- [ ] Documentation (clarifying comments, KDoc)
- [ ] Tests (Unit, Integration, UI tests)
- [ ] Other (please describe):

Put an `x` in all the boxes that apply.

## What's changed?
Describe with a few bullets **what's new:**
- Added a new Gradle plugin that can be used to upgrade the Gradle Wrapper (and create an PR if so)
- Added a GitHub Action workflow that will run periodically to check if there is an Wrapper upgrade
- 
_💡 Tip: Attach screenshots and screen recordings. It helps a lot!_

## Does this PR closes any GitHub Issues?

Check **[Ivy Wallet Issues](https://github.com/Ivy-Apps/ivy-wallet/issues)**.
- Closes #2449 

_Replace `{ISSUE_NUMBER}` with the id/number of the issue that you've fixed._

## Final Steps

Test your build again with the **app-demo.apk** generated by the `APK workflow`. This is an important step because it applies code obfuscation and R8 that often produces runtime exceptions which make the app crash.

> We don't have QA, you are the QA! That's why we require so much testing.

If everything still works fines, comment and tag **@ILIYANGERMANOV**. He'll try to merge your PR whenever possible.

**Thank you for your contribution! 🎉**

Some more notes:
You should replace the `secrets.GITHUB_TOKEN` with your own GitHub token.
Why?
Otherwise The GitHub Actions bot will create the PR with the wrapper upgrade.
This will work, but "test" Actions (CI) will not run because of this.
So it is better to replace this with your personal token.
With that, it looks like "you" created the PR and tehrefore the test Actions (CI) will run.

For testing purpose you can also downgrade Gradle (to 8.2) and then run locally:
```
WRAPPER_UPGRADE_GIT_TOKEN=[YOUR_TOKEN] ./gradlew upgradeGradleWrapperIvyWallet
```

### CODE REVIEW

File: .github/workflows/upgrade-gradle-wrapper.yml
Comment (ILIYANGERMANOV) [inline .github/workflows/upgrade-gradle-wrapper.yml:6]: What does this line mean?
How often would the cron run? 
Is it free for open-source repos?

**Review by ILIYANGERMANOV — COMMENTED** [review]
I don't feel very confident provide my personal GitHub PAT to a plugin. Is there another way to make it work?

Can you expand on the implications if we don't do that. If the bot opens the PR with the Bot username what's the issue? I don't understand why It'll use th wrong wrapper.

**Comment by ILIYANGERMANOV** [comment]
If, I understand correctly if it's not with my token - can I manually approve the CI run? Like I just did for your - if that's the case, it's perfect IMO 

Overall, your PR LGTM - sorry for the stupid questions but I don't fully understand the token part 😀

File: .github/workflows/upgrade-gradle-wrapper.yml
Comment (StefMa) [reply .github/workflows/upgrade-gradle-wrapper.yml:6]: This action runs every day at 8 Am UTC.
Yes it is free for open source projects to run scheduled actions.
However it runs only on "maintained"/"active" repositories. In case there is no "action" happening in this repository for longer than 30 days, this action will stop.

You're free to schedule this like you want 🙃. Maybe once a week is also fine. Or every two weeks 🤷‍♂️ your call ☺

**Comment by StefMa** [comment]
Hehe, no worries 😉
"An" token is required to create a pull request. Otherwise the plugin can't create it and will fail.

It doesn't matter "which" token you add, as long as the given token has write PR access to the repository, it works.

Therfore, the current code works! It will open an PR by the github-actions[bot] user.
And this is a "problem". For whatever reasons, all actions that will be triggered by an PR/commit or whatever will not be run in case the PR/commit was made by the github bot.

Honestly, I didn't find a way to manually trigger those actions afterwards. They don't run and you can't manually trigger approve them (like you did in this PR). The commit/PR will simply not tested.

Does this make sense? 🤔 
I hope this clarifies a bit more. 

I am not sure if you are aware of it, but you can simply create a new token with just write PR write access and add it to the github actions secrets. I personally don't see the reason why someone don't want to do this. Name it like WRAPPER_UPGRADE_TOKEN 🤷‍♂️.

However, if you don't want to do this, feel free to merge this PR as it is. PRs will still be created. But you should manually checkout the PR and do your smoke tests on it. I think this is also not a big deal as Gradle is the build tool. So either it builds the project or not 🤷‍♂️

File: .github/workflows/upgrade-gradle-wrapper.yml
Comment (ILIYANGERMANOV) [reply .github/workflows/upgrade-gradle-wrapper.yml:6]: Got it - thanks for the good explanation! 8 days sounds good for me 💯

**Comment by ILIYANGERMANOV** [comment]
> Hehe, no worries 😉 "An" token is required to create a pull request. Otherwise the plugin can't create it and will fail.
> 
> It doesn't matter "which" token you add, as long as the given token has write PR access to the repository, it works.
> 
> Therfore, the current code works! It will open an PR by the github-actions[bot] user. And this is a "problem". For whatever reasons, all actions that will be triggered by an PR/commit or whatever will not be run in case the PR/commit was made by the github bot.
> 
> Honestly, I didn't find a way to manually trigger those actions afterwards. They don't run and you can't manually trigger approve them (like you did in this PR). The commit/PR will simply not tested.
> 
> Does this make sense? 🤔 I hope this clarifies a bit more.
> 
> I am not sure if you are aware of it, but you can simply create a new token with just write PR write access and add it to the github actions secrets. I personally don't see the reason why someone don't want to do this. Name it like WRAPPER_UPGRADE_TOKEN 🤷‍♂️.
> 
> However, if you don't want to do this, feel free to merge this PR as it is. PRs will still be created. But you should manually checkout the PR and do your smoke tests on it. I think this is also not a big deal as Gradle is the build tool. So either it builds the project or not 🤷‍♂️

Thanks for the detailed explanation! Yeah, it clarifies things 👍 I'll just create a fine-grained token and scope it Ivy Wallet open PRs and it should work fine. Biggest security risk is the bot spamming the repo with PRs but that's fine - I can invalidate the token.

Sounds good! Let's do it. I'll create the `GRADLE_WRAPPER_UPGRADE_TOKEN` can you use it in the PR and we'll merge it. Great job @StefMa 💯

File: .github/workflows/upgrade-gradle-wrapper.yml
Comment (ILIYANGERMANOV) [inline .github/workflows/upgrade-gradle-wrapper.yml:21]: Sorry, one final thing, how is this supposed to work? Can you link me to the plugin docs? It seems to me that's exposing the token in the URL. Is that intended?

Comment (ILIYANGERMANOV) [reply .github/workflows/upgrade-gradle-wrapper.yml:21]: Nvm, all clear

**Review by ILIYANGERMANOV — APPROVED** [review]
All good - I did a little research and everything seems good to me! Thank you for this awesome automation - I'll merge it and test it out 👍

File: .github/workflows/upgrade-gradle-wrapper.yml
Comment (StefMa) [reply .github/workflows/upgrade-gradle-wrapper.yml:21]: Just for reference.
This the url to the plugin:
https://github.com/gradle/wrapper-upgrade-gradle-plugin

There is no official doc how to create the action. And honesty, I also don't know exactly why this has to be like this 😔. I guess this has something to do with how the plugin clones the project with got...

This is how it looks like in action:

**Pull request merged**
